client/client.go

```diff
@@ -102,7 +109,38 @@ func NewApiClient(conf Config) (*ApiClient, error) {
 	return New(conf)
 }
 
-// Client initialization
+// keepaliveDialer is used to set keepalive options on the socket used for
+// connection. This is for handling connection timeouts while server
+// is not accessible after connection was alrady started.
+type keepaliveDialer struct {
+	net.Dialer
+}
```

Comment:
Actually, I didn't even know that it was possible to be that granular, and the one interval you mention is what I would have expected from the API. I reckon that should be good enough, I mean, I don't think we don't need super fine grained control here, it's just a mechanism to time out in a roughly given time.